core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/Nulls.kt

```diff
@@ -56,94 +126,350 @@ internal inline val Float?.isNA: Boolean get() = this == null || this.isNaN()
 
 // region fillNaNs
 
-public fun <T, C> DataFrame<T>.fillNaNs(cols: ColumnsSelector<T, C>): Update<T, C> =
-    update(cols).where { it.isNaN }
-
-public fun <T> DataFrame<T>.fillNaNs(vararg cols: String): Update<T, Any?> =
-    fillNaNs { cols.toColumns() }
-
-public fun <T, C> DataFrame<T>.fillNaNs(vararg cols: KProperty<C>): Update<T, C> =
-    fillNaNs { cols.toColumns() }
-
-public fun <T, C> DataFrame<T>.fillNaNs(vararg cols: ColumnReference<C>): Update<T, C> =
-    fillNaNs { cols.toColumns() }
+/**
+ * ## The Fill NaNs Operation
+ *
+ * Replaces [`NaN`][NaN] values with given value or expression.
+ * Specific case of [update].
+ *
+ * Check out the [`fillNaNs` Operation Usage][FillNaNs.Usage].
+ *
+ * For more information: {@include [DocumentationUrls.Fill.FillNaNs]}
+ */
+internal interface FillNaNs {
+
+    /** @include [Update.Usage] {@arg [Update.UpdateOperationArg] [fillNaNs][fillNaNs]} */
+    interface Usage
+}
 
-public fun <T, C> DataFrame<T>.fillNaNs(cols: Iterable<ColumnReference<C>>): Update<T, C> =
-    fillNaNs { cols.toColumnSet() }
+/** {@arg [SelectingColumns.OperationArg] [fillNaNs][fillNaNs]} */
+internal interface SetFillNaNsOperationArg
+
+/**
+ * @include [FillNaNs] {@comment Description of the fillNaNs operation.}
+ * @include [LineBreak]
+ * @include [Update.Columns] {@comment Description of what this function expects the user to do: select columns}
+ * ## This Fill NaNs Overload
+ */
+private interface CommonFillNaNsFunctionDoc
+
+/**
+ * @include [CommonFillNaNsFunctionDoc]
+ * @include [SelectingColumns.Dsl.WithExample] {@include [SetFillNaNsOperationArg]}
+ * @include [Update.DslParam]
+ */
+public fun <T, C> DataFrame<T>.fillNaNs(columns: ColumnsSelector<T, C>): Update<T, C> =
+    update(columns).where { it.isNaN }
+
+/**
+ * @include [CommonFillNaNsFunctionDoc]
+ * @include [SelectingColumns.ColumnNames.WithExample] {@include [SetFillNaNsOperationArg]}
+ * @include [Update.ColumnNamesParam]
+ */
+public fun <T> DataFrame<T>.fillNaNs(vararg columns: String): Update<T, Any?> =
+    fillNaNs { columns.toColumns() }
+
+/**
+ * @include [CommonFillNaNsFunctionDoc]
+ * @include [SelectingColumns.KProperties.WithExample] {@include [SetFillNaNsOperationArg]}
+ * @include [Update.KPropertiesParam]
+ */
+public fun <T, C> DataFrame<T>.fillNaNs(vararg columns: KProperty<C>): Update<T, C> =
+    fillNaNs { columns.toColumns() }
+
+/**
+ * @include [CommonFillNaNsFunctionDoc]
+ * @include [SelectingColumns.ColumnAccessors.WithExample] {@include [SetFillNaNsOperationArg]}
+ * @include [Update.ColumnAccessorsParam]
+ */
+public fun <T, C> DataFrame<T>.fillNaNs(vararg columns: ColumnReference<C>): Update<T, C> =
+    fillNaNs { columns.toColumns() }
+
+/**
+ * TODO this will be deprecated
```

Comment:
Oh, sorry, all these stuff will be deprecated in the https://github.com/Kotlin/dataframe/pull/320
Ignore my comments about deprecation under the Iterables